java/arcs/core/testutil/HandleHelpers.kt

```diff
@@ -0,0 +1,106 @@
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.testutil
+
+import arcs.core.entity.Entity
+import arcs.core.entity.Handle
+import arcs.core.entity.QueryCollectionHandle
+import arcs.core.entity.ReadCollectionHandle
+import arcs.core.entity.ReadSingletonHandle
+import arcs.core.entity.ReadableHandle
+import arcs.core.entity.Reference
+import arcs.core.entity.Storable
+import arcs.core.entity.WriteCollectionHandle
+import arcs.core.entity.WriteSingletonHandle
+import kotlinx.coroutines.joinAll
+import kotlinx.coroutines.withContext
+
+/** Calls [Handle.close] with the handle's dispatcher context. */
+suspend fun <H : Handle> H.dispatchClose() = withContext(dispatcher) { close() }
+
+/** Calls [ReadableHandle.createReference] with the handle's dispatcher context. */
+suspend fun <H : ReadableHandle<U>, U, E : Entity>
+            H.dispatchCreateReference(entity: E): Reference<E> =
+    withContext(dispatcher) { createReference(entity) }
+
+/** Calls [ReadSingletonHandle.fetch] with the handle's dispatcher context. */
+suspend fun <H : ReadSingletonHandle<T>, T> H.dispatchFetch(): T? =
+    withContext(dispatcher) { fetch() }
+
+/**
+ * Calls [WriteSingletonHandle.store] with the handle's dispatcher context and waits for it to
+ * complete (including notifications being sent to other handles reading from the same store).
+ */
+suspend fun <H : WriteSingletonHandle<T>, T> H.dispatchStore(element: T) {
+    withContext(dispatcher) { store(element) }.join()
+    getProxy().waitForIdle()
+}
+
+/**
+ * Calls [WriteSingletonHandle.clear] with the handle's dispatcher context and waits for it to
+ * complete (including notifications being sent to other handles reading from the same store).
+ */
+suspend fun <H : WriteSingletonHandle<T>, T> H.dispatchClear() {
+    withContext(dispatcher) { clear() }.join()
+    getProxy().waitForIdle()
+}
+
+/** Calls [ReadCollectionHandle.size] with the handle's dispatcher context. */
+suspend fun <H : ReadCollectionHandle<T>, T> H.dispatchSize(): Int =
+    withContext(dispatcher) { size() }
+
+/** Calls [ReadCollectionHandle.isEmpty] with the handle's dispatcher context. */
+suspend fun <H : ReadCollectionHandle<T>, T> H.dispatchIsEmpty(): Boolean =
+    withContext(dispatcher) { isEmpty() }
+
+/** Calls [ReadCollectionHandle.fetchAll] with the handle's dispatcher context. */
+suspend fun <H : ReadCollectionHandle<T>, T> H.dispatchFetchAll(): Set<T> =
+    withContext(dispatcher) { fetchAll() }
+
+/**
+ * Calls [WriteCollectionHandle.store] with the handle's dispatcher context and waits for it to
+ * complete (including notifications being sent to other handles reading from the same store).
+ *
+ * This allows multiple elements to be stored and will wait until all the operations are done.
+ */
+suspend fun <H : WriteCollectionHandle<T>, T> H.dispatchStore(first: T, vararg rest: T) {
```

Comment:
Why did you go with `first, rest` over just one `vararg` argument?

Reply:
Because varags can be empty, and I didn't want `dispatchStore()` to compile.